src/DBFTPlugin/Consensus/ConsensusContext.Get.cs

```diff
@@ -56,9 +56,16 @@ private PreparationPayloadCompact GetPreparationPayloadCompact(ExtensiblePayload
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public byte GetPrimaryIndex(byte viewNumber)
+        public byte GetPriorityPrimaryIndex(byte viewNumber)
         {
-            int p = ((int)Block.Index - viewNumber) % Validators.Length;
+            int p = ((int)Block[0].Index - viewNumber) % Validators.Length;
+            return p >= 0 ? (byte)p : (byte)(p + Validators.Length);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public byte GetFallbackPrimaryIndex(byte viewNumber)
+        {
+            int p = ((int)Block[0].Index - viewNumber + 1) % Validators.Length;
```

Comment:
This will be changed to another formula that is more efficient.
It is described on the .pdf specification https://github.com/NeoResearch/dbft3-specification/blob/master/docs/dBFT3.0_Specification.pdf

Reply:
Done.

Reply:
@igormcoelho, we still need to double check the new formula.